File: testdata/usage_fixtures/java/UsageSites.java
// Java call sites whose API names the usage (ADR-0030).
package usagesites;

import javax.crypto.Cipher;

public class UsageSites {
    // WRAP_MODE is KEY TRANSPORT, not bulk encryption -- and the mode is the
    // only thing that says so, since the transformation string is identical.
    public Cipher wrap() throws Exception {
        Cipher cipher = Cipher.getInstance("RSA/ECB/OAEPWithSHA-256AndMGF1Padding");
        cipher.init(Cipher.WRAP_MODE, (java.security.Key) null);
        return cipher;
    }
}

File: testdata/usage_fixtures/go/usage_sites.go
// Go call sites whose API names the usage (ADR-0030).
package usagesites

import (
	"crypto/rand"
	"crypto/rsa"
	"crypto/sha256"
)

// EncryptOAEP is KEY TRANSPORT: RSA wrapping a symmetric key.
func WrapSessionKey(pub *rsa.PublicKey, sessionKey []byte) ([]byte, error) {
	return rsa.EncryptOAEP(sha256.New(), rand.Reader, pub, sessionKey, nil)
}

// VerifyPKCS1v15 VERIFIES.
func CheckSignature(pub *rsa.PublicKey, digest, sig []byte) error {
	return rsa.VerifyPKCS1v15(pub, 0, digest, sig)
}
